roxctl/scanner/generate/generate_test.go

```diff
@@ -0,0 +1,41 @@
+package generate
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/apiparams"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/istioutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestScannerGenerateValidation(t *testing.T) {
+	t.Run("not supported Istio version", func(t *testing.T) {
+		cmdBadIstio := scannerGenerateCommand{apiParams: apiparams.Scanner{IstioVersion: "0.1.0"}}
+
+		expectedErr := errox.NewErrInvalidArgs(fmt.Sprintf(
+			"unsupported Istio version %q used for argument %q. Use one of the following: [%s]",
+			"0.1.0", "--"+istioSupportArg, strings.Join(istioutils.ListKnownIstioVersions(), "|"),
+		))
+		actualErr := cmdBadIstio.validate()
+
+		require.Error(t, actualErr)
+		assert.ErrorIs(t, actualErr, errox.InvalidArgs)
+		assert.EqualError(t, actualErr, expectedErr.Error())
```

Comment:
Let's not check exact strings when comparing errors in tests — they are anyway always copy pasted from the prod code. Instead, please only check the expected error class via `ErrorIs()`.

Reply:
✅ I have refactored this part to test only relevant part of error message.

Reply:
I have mixed feelings about this on one hand I understand that checking error message is not the best idea but on the other in `roxctl` we present error to the end user so maybe it makes sens to assert on what is printed especially with errox libs that adds some prefixes to messages.